src/twisted/internet/defer.py

```diff
@@ -160,6 +166,65 @@ def maybeDeferred(f, *args, **kw):
 
 
 
+def unbox(deferred, raises=True, passthrough=True):
+    """
+    Unbox a L{Deferred} that has already fired its result.
+
+    Take a deferred that has already fired its result and unbox it, returning
```

Comment:
What exactly happens to the callback chain is currently sort of up for discussion? For my use case I want the result consumed and the deferred is finished (and if we don't at least consume the failure cases we get an ugly output when the Deferred gets garbage collected). However @Julian had a use case where he just wanted to "peek" into the deferred and not munge the results of it at all. 

In addition when I was implementing ``successResultOf`` and ``failureResultOf`` in terms of this (which is already done in this PR btw) I needed a variety of behaviors so I ended up making this function take two boolean arguments to control what happens to the result and what happens if it's an error.

Obviously given pretty much any behavior (except ``raises=True`` being the only behavior) you can implement the other behaviors as a wrapper around this, however that then loses some of the benefit of providing this function in the first place I think.

I'm not sure though, for me I think ``raises=True, passthrough=False`` because that makes "incorrect" values for ``raises`` / ``passthrough`` more obvious and I think it matches the most common use case for this function best (wanting to turn ``Deferred`` using code into a sync function).